source/blog/_posts/2018-08-23-against-microservices.md

```diff
@@ -0,0 +1,136 @@
+
+By giving the highest degree of freedom to your teams the structure of your software changes, too. Dependencies in such a huge product can no longer be top down tree like organized but are in fact more like a **bee hive**. Microservices can potentially be even more then that. The hive is multinational! Your services can be written in any language, come from any source and as long as network communication is possible you are fine. In extreme cases there can and will be C code communicating happily with a PHP script that notifies a Python app that provides data for Node app. Every one of these apps is self contained and uses a different set of tools for its build, deployment, integration and monitoring. Yeah!
+
+Now imagine something is broken. How can you inspect, debug and fix any issue in such a system? You need conventions!
+
+> "With great power comes great responsibility" - B. Parker
+
+If you want to base each new service on different technology you will find that your Microservices have multiple different solutions to the same common problem. [*Not invented here*](https://en.wikipedia.org/wiki/Not_invented_here) is the buildin result from this approach to team organizations.
+
+To keep with the bee hive metaphor: Even they grow over time from just a founding queen.
+
+In the end the freedom simply does not come for free. Where does scale outweigh central technical decision management? Some huge companies that develop concurrently on a single product of course have good reasons to go into that direction. And in the wild you will find scaled back solutions (e.g. single programming language, central logging and monitoring, centrally enforced architecture, even single process) that try to get the pros without the cons. But what is left in these cases?
+
+
+## Problems arise
+
+I get hyped by talks on conferences. Either watching them directly live or in recording. But Microservices tend to tickle a nerve where I constantly ask myself "What aren't they talking about?" Microservices add pressure on sophisticated technological solutions where a monolith can get away with far less effort. Take logging for example. While a single process application may be inspectable through a file log initially, a distributed system can not.
+
+Or how do you secure that the negaotiated API interfaces actually work? Integration tests of course! All neighboring services must be integrated with one another, through the network, which means lots of tests for lots of services with lots of communication. 
+
+How will developers set up the application locally? Docker containers of course! Each Service runs in it's own container, comes usually at least one storage container. Somewhere is a routing configuration.
+
+Scaling contrary to common belief is an issue, too. There may be many independent Services that need to be scaled appropriately to improve the overall Performance of the application. 
```

Comment:
Sacling __,__ contrary to common belief __,__ is an issue --,-- too.
independent __s__ervices __,__
overall performance
->(imho) You are talking about a scale hardly achiveable by a monolith, already.